src/business-logic/services/test/throttled-rundown-service.spec.ts

```diff
@@ -0,0 +1,125 @@
+
+        const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+        await testee.takeNext(anything())
+
+        const result: () => Promise<void> = () => testee.takeNext(anything())
+
+        expect(result).toThrow(ThrottledRundownException)
+      })
+
+      it('throws error when two different actions are performed with 0ms delay', async () => {
+        const mockRundownService: RundownService = mock<RundownService>()
+
+        const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+        await testee.activateRundown(anything())
+
+        const result: () => Promise<void> = () => testee.takeNext(anything())
+
+        expect(result).toThrow(ThrottledRundownException)
+      })
+
+      it('throws error when two take nexts are performed with 499ms delay', async () => {
+        const mockRundownService: RundownService = mock<RundownService>()
+        const now: number = Date.now()
+
+        jest.setSystemTime(now)
+
+        const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+        await testee.takeNext(anything())
+
+        jest.advanceTimersByTime(499)
+
+        const result: () => Promise<void> = () => testee.takeNext(anything())
+
+        expect(result).toThrow(ThrottledRundownException)
+      })
+    })
+  })
+
+  describe('when two operations are performed outside of a throttled interval', () => {
+    it('performed two take nexts with 500ms between', async () => {
+      const mockRundownService: RundownService = mock<RundownService>()
+      const now: number = Date.now()
+      jest.setSystemTime(now)
+
+      const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+      await testee.takeNext(anything())
+
+      jest.advanceTimersByTime(500)
+
+      await testee.takeNext(anything())
+
+      verify(mockRundownService.takeNext(anything())).times(2)
+    })
+
+    it('performed two different operations with 500ms between', async () => {
+      const mockRundownService: RundownService = mock<RundownService>()
+      const now: number = Date.now()
+
+      jest.setSystemTime(now)
+
+      const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+      await testee.activateRundown(anything())
+
+      jest.advanceTimersByTime(500)
+
+      await testee.takeNext(anything())
+
+      verify(mockRundownService.takeNext(anything())).once()
+    })
+  })
+
+  describe('when performed operations witch should not have throttled mechanism', () => {
+    it('performed two insert part as next operations without any time between', async () => {
+      const mockRundownService: RundownService = mock<RundownService>()
+
+      const testee: ThrottledRundownService = new ThrottledRundownService(instance(mockRundownService))
+
+      await testee.insertPartAsNext(anything(), anything())
```

Comment:
`anything()` is a matcher object for mocking, not a value that should be used on real instances. Instead, we should use correctly typed values. You can use the `TestEntityFactory` if you need to create entity data for the tests.

Reply:
All anything() was removed.